standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -2814,21 +2649,12 @@ long getPartsFound() {
     catName = StringUtils.normalizeIdentifier(catName);
     dbName = StringUtils.normalizeIdentifier(dbName);
     tblName = StringUtils.normalizeIdentifier(tblName);
-    if (!shouldCacheTable(catName, dbName, tblName) || (canUseEvents && rawStore.isActiveTransaction())) {
+    if (!shouldCacheTable(catName, dbName, tblName) || (canUseEvents && rawStore.isActiveTransaction()) || !sharedCache
+        .isTableConstraintValid(catName, dbName, tblName)) {
       return rawStore.getCheckConstraints(catName, dbName, tblName);
     }
+    return sharedCache.listCachedCheckConstraint(catName, dbName, tblName);
```

Comment:
below check was present earlier:

```
if (CollectionUtils.isEmpty(keys)) {	
      return rawStore.getCheckConstraints(catName, dbName, tblName);	
}
```

Why is this removed? Is this check moved somewhere else?

Reply:
Added flag - isConstraintValid to check null value is consistent instead of going to raw store.